### PR TITLE
Implement the change from eWASM to ewasm

### PR DESCRIPTION
**Pull request by jakelang**
As we have discussed, the styling of the name "eWASM" should be changed to "ewasm". The exception made here is in the cases when ewasm is used at the start of the sentence and when used in an acronym such as ECI, for which the first letter will be capitalized as with a normal word.

### CODE REVIEW

**Comment by lrettig** [comment]
For the record my complaint with the (always-lowercase) "ewasm" is that it looks silly when writing things like "EVM and ewasm" -- I definitely prefer "EVM and Ewasm". It is a proper noun.

**Comment by jakelang** [comment]
Updated the README.

File: contract_interface.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# eWASM Contract Interface (ECI) Specification (*Revision 0*)
+# Ewasm Contract Interface (ECI) Specification (*Revision 0*)
 
-The eWASM Contract Interface (ECI) specifies the structure of a contract module.
+The Ewasm Contract Interface (ECI) specifies the structure of a contract module.
```
Comment (axic) [inline contract_interface.md]: So is it ewasm or Ewasm?

Comment (jakelang) [reply contract_interface.md]: As I noted in the PR, I figured that the name should be treated as a common noun but stylized with lowercase. So in titles, acronyms, and at the beginning of sentences I prefer Ewasm but in other cases it is ewasm.

**Comment by chfast** [comment]
I was neutral to the naming issue, but realized that the _E_ in _Ewasm_ is more like _E_ in _Ethereum_ while _e_ in _ewasm_ is more like _e_ in the _Internet Explorer_ logo.

**Comment by jakelang** [comment]
@chfast Now that you mention it, it actually may be better to treat Ewasm as a proper noun for both the reason above and the fact that grammatically it is already treated as a proper noun while the initial decision capitalizes it as a common noun.

**Comment by jakelang** [comment]
Rebased.

**Pull request closed without merging**